lib/puppet/provider/jbosscli.rb

```diff
@@ -116,9 +116,23 @@ def self.last_execute_status
   def self.execshell(cmd)
     `#{cmd}`
   end
-  
+
+  def self.reload_facter
+    Facter.clear
```

Comment:
I dont like the need of clearing all Facter facts. This is not elegant. Mayby there is a better way? Mayby reuse code for fact and for provider?

If you must clear some facts, please you something like that: https://github.com/coi-gov-pl/puppet-jboss/blob/develop/spec/unit/facter/jboss_configfile_spec.rb#L12-L13